staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -114,6 +120,7 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	if handlingInfo.local {
 		if r.localDelegate == nil {
 			http.Error(w, "", http.StatusNotFound)
+			httpStatus = http.StatusNotFound
 			return
 		}
 		r.localDelegate.ServeHTTP(w, req)
```

Comment:
Where is the HTTP status written by the CR handler captured? It looks like, as currently implemented, the metric will report 200 regardless of the status written by the delegate handler.

Reply:
The same comment applies to the case where the response status is set by the upgrade aware proxy handler.

Reply:
@benluddy The latest modification reads the status of http.ResponseWriter to record the status of the delegate handler. Please review again if you have time.